python/paddle/distributed/auto_parallel/converter.py

```diff
@@ -0,0 +1,449 @@
+                                cur_name, pre_name))
+                        param_match_with_pre.append(cur_name)
+                        param_match_with_cur.append(pre_name)
+                        break
+                break
+
+        return params_dict, param_match_with_pre, param_match_with_cur
+
+    @staticmethod
+    def merge_and_slice(param_list, pre_dist_attr, cur_dist_attr):
+        """
+        Merge parameters with previous dist_attr and slice parameters with current dist_attr
+
+        Returns:
+            param(numpy.narray): a parameter's value of current rank.
+        """
+        assert isinstance(param_list, list)
+        assert all(isinstance(p, np.ndarray) for p in param_list)
+
+        if pre_dist_attr == cur_dist_attr:
+            # skip merge and slice param
+            rank_id = paddle.distributed.get_rank()
+            index = cur_dist_attr["process_group"].index(rank_id)
+            param = param_list[index]
+        else:
+            pre_dims_mapping = pre_dist_attr["dims_mapping"]
+            cur_dims_mapping = cur_dist_attr["dims_mapping"]
+            if len(set(pre_dims_mapping)) > 1 or -1 not in pre_dims_mapping:
+                # merge param
+                param = Converter.merge_with_dist_attr(param_list,
+                                                       pre_dist_attr)
+            else:
+                # skip merge param
+                param = param_list[0]
+
+            if len(set(cur_dims_mapping)) > 1 or -1 not in cur_dims_mapping:
+                # slice param
+                param = Converter.slice_with_dist_attr(param, cur_dist_attr)
+
+        return param
+
+    @staticmethod
+    def merge_with_dist_attr(param_list, dist_attr):
+        """ Merge param with distributed attribute """
+        from .reshard import _compute_complete_shape, _compute_partition_index
+
+        dims_mapping = dist_attr["dims_mapping"]
+        process_shape = dist_attr["process_shape"]
+        process_group = dist_attr["process_group"]
+        # get the complete shape of the parameter
+        complete_shape = _compute_complete_shape(param_list[0].shape,
+                                                 process_shape, dims_mapping)
+        # merge the parameter with dist_attr
+        partition_param_list = []
+        merged_partiton = []
+        for process in process_group:
+            partition_index = _compute_partition_index(
+                process, complete_shape, dims_mapping, process_shape,
+                process_group)
+            index = process_group.index(process)
+            if partition_index not in merged_partiton:
+                merged_partiton.append(partition_index)
+                Converter.merge(partition_param_list, param_list[index],
+                                partition_index, complete_shape)
+
+        assert len(partition_param_list) == 1 or not partition_param_list, \
+            "Fail to merge parameter"
```

Comment:
also throw out the information of failed tensor, like name, pre_dist_attr, target_dist_attr

Reply:
done.